### PR TITLE
Change fields used for custom view blocks in layouts.

### PR DESCRIPTION
**Pull request by Graham-72**
As proposed in issue #2723 - Clarify where layouts obtains description for a views block.

### CODE REVIEW

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 1]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
    */
   function execute_hook_block_list($delta = 0, $edit = array()) {
     $delta = $this->view->name . '-' . $this->display->id;
-    $info = $this->get_option('block_description');
+    $block_title = $this->get_option('block_description');
+		$block_description = $this->get_option('block_layout_description');
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc]: line needs +1 space indentation

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 2]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
...
+        $block_title = t('View: @view', array('@view' => t($this->view->get_human_name())));
       }
       else {
-        $info = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
+        $block_title = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
       }
     }
+    else {
+		  $block_title = t('View: ').$block_title;
+		}
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc]: line needs +1 space indentation

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 3]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
...
-        $info = t('View: @view', array('@view' => t($this->view->get_human_name())));
+        $block_title = t('View: @view', array('@view' => t($this->view->get_human_name())));
       }
       else {
-        $info = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
+        $block_title = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
       }
     }
+    else {
+		  $block_title = t('View: ').$block_title;
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc]: - spaces before/after the dot
- I think that this should be `t('View block: ')` or `t('View display: ')`

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 4]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
    */
   function execute_hook_block_list($delta = 0, $edit = array()) {
     $delta = $this->view->name . '-' . $this->display->id;
-    $info = $this->get_option('block_description');
+    $block_title = $this->get_option('block_description');
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc]: perhaps call this variable `$block_admin_title` to match the label in the views UI. That's what I meant when I was saying that we should rename the variables.

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 5]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
...
 
-    if (empty($info)) {
+    if (empty($block_title)) {
       if ($this->display->display_title == $this->definition['title']) {
-        $info = t('View: @view', array('@view' => t($this->view->get_human_name())));
+        $block_title = t('View: @view', array('@view' => t($this->view->get_human_name())));
       }
       else {
-        $info = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
+        $block_title = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc:81]: This entire if/else is pointless. We are repeating the entire thing just to append the display title.

File: core/modules/views/plugins/views_plugin_display_block.inc (hunk 4 above)
Comment (Graham-72) [reply core/modules/views/plugins/views_plugin_display_block.inc]: It is unfortunate that this has been named block_description in config - it creates confusion.

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 6]
```diff
@@ -70,20 +70,24 @@ class views_plugin_display_block extends views_plugin_display {
...
     }
+    else {
+		  $block_title = t('View: ').$block_title;
+		}
     return array(
       $delta => array(
-        'info' => $info,
-        'description' => $this->view->description ? check_plain(t($this->view->description)) : t('Block display from the "@view" view.', array('@view' => t($this->view->get_human_name()))),
+        'info' => $block_title,
+        'description' => check_plain(t($block_description)) .' ('. t('A block display from the "@view" view.', array('@view' => t($this->view->get_human_name()))) .')',
```
Comment (klonos) [inline core/modules/views/plugins/views_plugin_display_block.inc]: codding standards: space before/after each dot.

File: core/modules/views/plugins/views_plugin_display_block.inc (hunk 5 above)
Comment (Graham-72) [reply core/modules/views/plugins/views_plugin_display_block.inc:81]: @klonos I am leaving this for a later amendment.

**Comment by Graham-72** [comment]
I do not understand why zenci is reporting failed tests. I do not get these when using Backdrop's Testing module.

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 7]
```diff
@@ -125,15 +134,25 @@ class views_plugin_display_block extends views_plugin_display {
...
+      $block_layout_description = t('None');
+    }
+    
     $options['block_description'] = array(
       'category' => 'block',
       'title' => t('Block admin label'),
       'value' => views_ui_truncate($block_description, 24),
     );
+    
+    $options['block_layout_description'] = array(
```
Comment (quicksketch) [inline core/modules/views/plugins/views_plugin_display_block.inc]: These internal keys are a mess (not your fault), perhaps this is an opportunity to change "block_description" to "block_admin_label" and we can make this new key "block_admin_description"? Such a change would require an update hook for existing views.

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 8]
```diff
@@ -70,20 +70,29 @@ class views_plugin_display_block extends views_plugin_display {
    */
   function execute_hook_block_list($delta = 0, $edit = array()) {
     $delta = $this->view->name . '-' . $this->display->id;
-    $info = $this->get_option('block_description');
-
-    if (empty($info)) {
+    $block_admin_title = $this->get_option('block_description');
```
Comment (quicksketch) [inline core/modules/views/plugins/views_plugin_display_block.inc]: Extra space here.

File: core/modules/views/plugins/views_plugin_display_block.inc [hunk 9]
```diff
@@ -70,20 +70,29 @@ class views_plugin_display_block extends views_plugin_display {
...
       else {
-        $info = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
+        // Block display has a name 'display title' so include this in display.
+        $block_admin_title = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
       }
     }
+    else {
+      // There is a block admin title so use this, prefixed with 'View'
+      // to indicate that this block originates from a view.
+      $block_admin_title = t('View: ') . $block_admin_title;
```
Comment (quicksketch) [inline core/modules/views/plugins/views_plugin_display_block.inc]: Not sure about this mandatory prefix ([see comment here](https://github.com/backdrop/backdrop-issues/issues/2723#issuecomment-309225750)), let's keep discussing in the issue.

Comment (Graham-72) [reply core/modules/views/plugins/views_plugin_display_block.inc]: I'm not sure about this either, and I prefer your tooltip suggestion. But I do find it helpful when looking in layouts at a lengthy list of blocks that many are prefixed 'field:' and some 'view:'.

**Pull request closed without merging**